Added infection and improved some tests

alfredbez opened the pull request.

flow-control commented on tests/Unit/DataType/BoolFilterTest.php:
```diff
@@ -0,0 +1,43 @@
+     * @param mixed $userInput
+     * @param mixed $returnValue
+     */
+    public function testReturnsGivenEqualValue($userInput, $returnValue): void
```
```suggestion
     */
    public function testReturnsGivenEqualValue(bool $userInput, bool $returnValue): void
```

flow-control commented on tests/Unit/DataType/FloatFilterTest.php:
```diff
@@ -21,6 +21,12 @@ public function testThrowsExceptionOnNoInput(): void
+    public function testGivesEqualParameterIfSet(): void
+    {
+        $filter = new FloatFilter(1.0);
+        $this->assertEquals(1.0, $filter->equals());
```
```suggestion
        $this->assertSame(1.0, $filter->equals());
```

flow-control commented on tests/Unit/Framework/RequestReaderTest.php:
```diff
@@ -118,4 +120,22 @@ public function testGetGraphQLRequestDataWithInputRequest(): void
+        $_REQUEST['query']         = 'query {token_}';
+        $_REQUEST['variables']     = '{"foo":"bar"}';
+        $_REQUEST['operationName'] = 'operation_name';
+        $this->assertEquals(
```
```suggestion
        $this->assertSame(
```

flow-control commented on tests/Unit/Framework/RequestReaderTest.php:
```diff
@@ -118,4 +120,22 @@ public function testGetGraphQLRequestDataWithInputRequest(): void
+            ],
+            $requestReader->getGraphQLRequestData()
+        );
+        unset($_SERVER['CONTENT_TYPE']);
```
```suggestion
        unset($_SERVER['CONTENT_TYPE']);
        unset($_REQUEST['query']);
        unset($_REQUEST['variables']);
        unset($_REQUEST['operationName']);
```

flow-control commented on .travis.yml:
```diff
@@ -30,3 +30,4 @@ script:
     - composer test:lint-fixer
     - composer test:static
     - composer test:unit
+    - composer test:infection
```
Could you somehow update this to only run on PHP >= 7.2? I think it is okay to ignore PHP 7.1 in this case

alfredbez replied:
I downgraded infection to an older version which is compatible with PHP 7.1

alfredbez commented:
I've fixed the open things and rebased it, build is green now for all PHP versions.

> One thing I will have a deeper look into is `tests/Unit/Controller/LoginTest.php`. It somehow bothers me ;-) Mainly I will investigate if there is a better option then to have these mocks.

I basically copied it from [here](https://github.com/OXID-eSales/graphql-base-module/blob/795b056ce5cf52221d4580875e66ccca28476c6d/tests/Unit/Service/AuthenticationServiceTest.php#L39-L46), maybe we can find a cleaner solution for both cases.

flow-control commented:
> > One thing I will have a deeper look into is `tests/Unit/Controller/LoginTest.php`. It somehow bothers me ;-) Mainly I will investigate if there is a better option then to have these mocks.
> 
> I basically copied it from [here](https://github.com/OXID-eSales/graphql-base-module/blob/795b056ce5cf52221d4580875e66ccca28476c6d/tests/Unit/Service/AuthenticationServiceTest.php#L39-L46), maybe we can find a cleaner solution for both cases.

Yeah I did this, and I dont like it ;-)

flow-control commented:
I don't like the alternative either :wink: 
LGTM and thanks again for this contribution!
